Formula/needle.rb

```diff
@@ -0,0 +1,15 @@
+class Needle < Formula
+  desc "Compile-time safe Swift dependency injection framework with real code"
+  homepage "https://github.com/uber/needle"
+  url "https://github.com/uber/needle.git",
+      :tag      => "v0.8.5",
+      :revision => "f15f0ea0249389ca95381b6150a07614df6a6472"
+
+  def install
+    system "make", "install", "BINARY_FOLDER_PREFIX=#{prefix}"
+  end
+
+  test do
+    system "#{bin}/needle", "version"
```

Comment:
We need a test that exercises the some of the functionality of the app.

Reply:
The functionality of the generator requires a source codebase to parse and destination to generate code into. This isn't really feasible. I followed the same approach as SourceKitten https://github.com/Homebrew/homebrew-core/blob/master/Formula/sourcekitten.rb

Reply:
There are examples where we compile source code. You can inline them or use resources ([example](https://github.com/Homebrew/homebrew-core/blob/master/Formula/flif.rb)). You can take a look at example [here](https://github.com/Homebrew/homebrew-core/blob/master/Formula/tinyxml2.rb). More information in [cookbook](https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula). I think you can even have sample git repo to clone in resources.

Reply:
Needle is meant to parse a codebase and generate a bunch of DI code. Is that something we really want to exercise during installation? That seems a bit much.

Reply:
Ideally we'd have a better test. Maybe run on the TicTacToe example from your repository?

Reply:
I couldn't find any documentation on how to pass the resource file's path into Needle. The way I have it setup right now doesn't pass in the right value. I'm getting `"No such file or directory"`.